packages/grid/_modules_/grid/hooks/features/export/useGridCsvExport.tsx

```diff
@@ -38,7 +38,7 @@ export const useGridCsvExport = (apiRef: GridApiRef): void => {
       return buildCSV({
         columns: exportedColumns,
         rows: visibleSortedRows,
-        selectedRowIds: selection,
+        selectedRowIds: Array.isArray(selection) ? selection : [selection],
```

Comment:
We could change below to use `gridArraySelectionStateSelector` and remove this if-is-an-array check.

https://github.com/mui-org/material-ui-x/blob/f79a1fd0c32ddc9a7d2e1cdf87e81d369ce6827e/packages/grid/_modules_/grid/hooks/features/export/useGridCsvExport.tsx#L20